Source/WebCore/page/Navigation.cpp

```diff
@@ -121,14 +124,21 @@ enum EventTargetInterfaceType Navigation::eventTargetInterface() const
     return EventTargetInterfaceType::Navigation;
 }
 
+static RefPtr<DOMPromise> createDOMPromise(Ref<DeferredPromise>& deferredPromise)
```

Comment:
This can likely be const &.

Reply:
Neither seems write. We only pass smart pointers to transfer ownership. So either this remains a smart pointer and we should use a r-value reference `Ref&&`, or this should be using a `DeferredPromise&`.